packages/target-decisioning-engine/src/allocationProvider.js

```diff
@@ -42,21 +58,13 @@ export function computeAllocation(
   visitorId,
   salt = CAMPAIGN_BUCKET_SALT
 ) {
-  // TODO: may want to memoize this
-
-  // Generate a unique visitor ID and persist it.
-  const vid = getOrCreateVisitorId(visitorId); // first non-blank marketingCloudVisitorId, tntId, thirdPartyId
-
   // Generate a device id based on visitorId, clientCode, campaignId and a salt value
-  const deviceId = `${clientId}.${activityId}.${vid}.${salt}`;
-
-  // Use the MurmurHash3 (32-bit) hashing algorithm to generate a numeric 10-digit hash based on the device id.
-  const output = MurmurHash3(deviceId).result();
+  const deviceId = [
+    clientId,
+    activityId,
+    typeof visitorId === "string" ? visitorId : getOrCreateVisitorId(visitorId),
```

Comment:
is visitorId guaranteed to never be an empty string? if not, we need to add an empty string check as well

Reply:
good idea, i changed it to check for `visitorId.length > 0` as well